plugins/jobs/src/js/JobsRunNow.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-unused-vars */
+import React from "react";
+/* eslint-enable no-unused-vars */
+
+import { graphqlObservable } from "data-service";
+import gql from "graphql-tag";
+
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/do";
+import "rxjs/observable/empty";
+
+import { defaultSchema } from "./JobsMenuClient";
+
+const runNowJobMutation = gql`
+  mutation {
+    runJob(jobId: $jobId) {
+      jobId
+    }
+  }
+`;
+
+const runNowJobGraphql = jobId => {
+  return graphqlObservable(runNowJobMutation, defaultSchema, jobId);
+};
+
+const runNowEvent$ = new Subject();
+
+export const jobsRunNow$ = runNowEvent$.switchMap(jobId => {
+  return runNowJobGraphql(jobId);
+});
+
+export const jobsRunNowAction = id => {
+  return {
+    label: "Run Now",
+    onItemSelect() {
+      jobsRunNow$.next({ jobId: id });
+    }
+  };
+};
```

Comment:
Here we could test that running the callback (onItemSelect()) really triggers the graphql call.

Reply:
When I run the method from the test, for some reason the .next doesn't make the graphql call. Will need to research this more.